src/condor_utils/compat_classad_util.h

```diff
@@ -93,6 +93,8 @@ bool ClassAdsAreSame( ClassAd *ad1, ClassAd * ad2, StringList * ignored_attrs=NU
 
 void CopyMachineResources(ClassAd &destAd, const ClassAd & srcAd, bool include_res_list);
 
+void CopySelectAttrs(ClassAd &destAd, const ClassAd &srcAd, std::string attrs);
```

Comment:
attrs should be a const string ref, not a deep copy of the string as you do here.